src/ir_Corona.cpp

```diff
@@ -0,0 +1,569 @@
+    return false;
+
+  uint16_t pos = 0;
+  uint16_t used = 0;
+
+  // Data Section #1 - 3 loop
+  // e.g.
+  //   bits = 56; bytes = 7;
+  // #1  *(results->state + pos) = {0x28, 0x61, 0x3D, 0x19, 0xE6, 0x37, 0xC8};
+  // #2  *(results->state + pos) = {0x28, 0x61, 0x6D, 0xFF, 0x00, 0xFF, 0x00};
+  // #3  *(results->state + pos) = {0x28, 0x61, 0xCD, 0xFF, 0x00, 0xFF, 0x00};
+  for (uint8_t section = 0; section < kCoronaAcSections; section++) {
+    DPRINT(uint64ToString(section));
+    used = matchGeneric(results->rawbuf + offset, results->state + pos,
+                        results->rawlen - offset, kCoronaAcBitsShort,
+                        kCoronaAcHdrMark, kCoronaAcHdrSpace,
+                        kCoronaAcBitMark, kCoronaAcOneSpace,
+                        kCoronaAcBitMark, kCoronaAcZeroSpace,
+                        kCoronaAcBitMark, kCoronaAcSpaceGap, true,
+                        _tolerance + kCoronaTolerance, kMarkExcess, false);
+    if (used == 0) return false;  // We failed to find any data.
+    // short versions section 0 is special
+    if (strict && !IRCoronaAc::validSection(results->state, pos,
+                                            isLong ? section : 3))
+      return false;
+    offset += used;  // Adjust for how much of the message we read.
+    pos += kCoronaAcSectionBytes;  // Adjust by how many bytes of data was read
+    // don't read more data then what we have
+    if (results->rawlen <= offset)
+      break;
+  }
+
+  // Re-check we got the correct size/length due to the way we read the data.
+  if (strict && pos * 8 != kCoronaAcBits && pos * 8 != kCoronaAcBitsShort) {
+    DPRINTLN("strict bit match fail");
+    return false;
+  }
+
+  // Success
+  results->decode_type = decode_type_t::CORONA_AC;
+  results->bits = pos * 8;
+  // No need to record the state as we stored it as we decoded it.
+  // As we use result->state, we don't record value, address, or command as it
+  // is a union data type.
+  return true;
+}
+#endif  // DECODE_CORONA_AC
+
+/// Class constructor for handling detailed Corona A/C messages.
+/// @param[in] pin GPIO to be used when sending.
+/// @param[in] inverted Is the output signal to be inverted?
+/// @param[in] use_modulation Is frequency modulation to be used?
+/// @return An IRCoronaAc object.
+IRCoronaAc::IRCoronaAc(const uint16_t pin, const bool inverted,
+                       const bool use_modulation)
+    : _irsend(pin, inverted, use_modulation) { stateReset(); }
+
+/// Reset the internal state to a fixed known good state.
+/// @note The state is powered off.
+void IRCoronaAc::stateReset(void) {
+  // known good state
+  remote_state[kCoronaAcSectionD0Pos] = kCoronaAcSectionD0Base;
+  remote_state[kCoronaAcSectionD1Pos] = 0x00;  // ensure no unset mem
+  setTemp(kCoronaAcMinTemp);
+  setMode(kCoronaAcModeCool);
+  setFan(kCoronaAcFanAuto);
+  setOnTimer(kCoronaAcTimerOff);
+  setOffTimer(kCoronaAcTimerOff);
+  checksum(remote_state);
```

Comment:
I don't think this line is needed. It's use in `getRaw()` covers all "external" access to `remote_state`. Your call.

Reply:
Agreed, left there to make sure internal state was known